### ISSUE TITLE
Path Of Exile hard freeze using Wine Staging Gallium Nine

### ISSUE THREAD

**Issue by forkbomb1** [iXit/Mesa-3D#296]
Since unknown version (haven't played a while) PoE freezes at either map loading screen or after up to minute of playing. The whole system becomes unresponsive and has to be hard-reset.
Haven't found anything of import in the logs.
I'm on OpenSuse Tumbleweed + Mesa 17.4-dev + Wine 2.21 Staging + Gallium Nine.

Other people are affected as well:
https://www.reddit.com/r/wine_gaming/comments/6xdb63/path_of_exile_with_215_gallium_nine_freezes_pc/

The game runs without Gallium Nine, albeit with subpar performance.

**Comment by oleerik** [comment]
I can confirm that I also experience this. Would be amazing to see this fixed.

**Comment by gemantzu** [comment]
I can confirm this issue as well (Solus - Wine 2.21 staging + Gallium Nine + Mesa 17.3)

**Comment by siro20** [comment]
Please always mention the card you are using. If on radeon please disable csmt using csmt_force=0.
Try to provide a full mesa debug log or backtrace.

**Comment by siro20** [comment]
Also an apitrace file that shows the issue is very helpful.

**Comment by gemantzu** [comment]
I just tested it again with the new wine-3.0-rc2 on Solus, kernel: 4.14.8-41.current. 
ixni -F:
`Graphics:  Card: Advanced Micro Devices [AMD/ATI] Tahiti XT [Radeon HD 7970/8970 OEM / R9 280X]
           Display Server: x11 (X.Org 1.19.5 )
           drivers: ati,amdgpu (unloaded: modesetting,fbdev,vesa,radeon)
           Resolution: 1920x1080@60.00hz, 1920x1080@60.00hz
           OpenGL: renderer: AMD Radeon HD 7900 Series (TAHITI / DRM 3.19.0 / 4.14.8-41.current, LLVM 5.0.0)
           version: 4.5 Mesa 17.3.1`

What can I provide you with so you can find out what's going on?

**Comment by john-tho** [comment]
Not sure if the same issue, but:
I was getting freeze on second game area loading screen using DirectX9Ex
Example:
PathOfExile would load into town and function
Upon loading to enter hideout, game would freeze.

Using DirectX9, or `PathOfExile.exe --nodx9ex` I did not experience my issue.

**Comment by fosspill** [comment]
--nodx9ex does not work for me, exact same full system freeze on both.
Exactly what logs would need to be provided to fix this issue?

```Graphics:  Card: Advanced Micro Devices [AMD/ATI] Vega 10 XT [Radeon RX Vega 64]
           Display Server: x11 (X.Org 1.19.6 ) drivers: amdgpu (unloaded: vesa) Resolution: 1920x1080@144.00hz
           OpenGL: renderer: Radeon RX Vega (VEGA10 / DRM 3.23.0 / 4.15.7-1-MANJARO, LLVM 5.0.1)
           version: 4.5 Mesa 17.3.6
```

**Comment by gemantzu** [comment]
Solus had mesa 18.0.0.rc for a few days and things somehow seemed to get even worse in that version. Idk why, but there was a rollback to 17.something.something, so will test again when 18 comes back.

**Comment by axeldavy** [comment]
A hard freeze is very bad, but is hardly coming from gallium nine (more likely from the underlying driver).
It may be a shader compilation problem.

Please reproduce the freeze and give the log produced with R600_DEBUG=vs,ps NINE_DEBUG=all NINE_TGSI_DUMP=TRUE NINE_FF_DUMP=TRUE
To have NINE_DEBUG=all do anything you need mesa built with --enable-debug.

This should produce a huge log with all the shader codes and the last calls will be able to identify the shader with the hang.

**Comment by fosspill** [comment]
Where would this log be saved?

**Comment by axeldavy** [comment]
R600_DEBUG=vs,ps NINE_DEBUG=all NINE_TGSI_DUMP=TRUE NINE_FF_DUMP=TRUE wine myapp.exe &> myhugelog

Then compress this file with maximum xz compression (xz -9) and post it on some big file sharing platform.

**Comment by gemantzu** [comment]
[Dropbox Link](https://www.dropbox.com/s/4e9w97xe3inrg13/myhugelog.tar.gz?dl=0)
I don't know if this is what you want... Enabling Gallium Nine, halts the game in loading screen.

**Comment by kmk3** [comment]
Hello, I had the same issue (system freeze, needing hard-reset) in the hideout
with mesa 17.2+ and radeonsi on Manjaro, but it seems to be resolved as of mesa
17.3.8.

I would suggest to try with the latest packages and see if it works again.

Upstream bug:
https://bugs.freedesktop.org/show_bug.cgi?id=103246

**Packages** (as of the last test):
linux416 4.16.4-1

lib32-llvm-libs 6.0.0-1
llvm-libs 6.0.0-4

lib32-mesa 18.0.1-1
lib32-mesa-vdpau 18.0.1-1
libva-mesa-driver 18.0.1-1
mesa 18.0.1-1
mesa-vdpau 18.0.1-1

wine-gaming-nine 3.7-1

**Comment by montvid** [comment]
Does not seem to work with padoka ppa unstable mesa 19.1 git. Some areas load but some don't . Wine 4.0 4.1 staging emulation crashes the game, gallium nine does not - - the game disconnects from the server back to game intro screen.

Can anyone confirm that PoE works with gallium nine in dx9 and what wine version?

**Comment by gemantzu** [comment]
Coming next patch, ggg will drop support for DX9 and WinXP, so we will probably have only DXVK to work with. Gallium nine only works with dx9, right? Source: [PoE Announcement](https://www.pathofexile.com/forum/view-thread/2319193)

**Comment by montvid** [comment]
@gemantzu well it is interesting enough why gallium nine(g9) does not like to work with PoE. If it is a bug in g9 it could improve other games too.

**Comment by gemantzu** [comment]
@montvid I think that it has to do with their engine (custom made), let's see how things play out now that they will ditch dx9 and dx11 is the only option.

**Comment by kmk3** [comment]
For me, it is back to freezing the entire computer after walking for a few
seconds.  I tried for weeks to make it work, but I found it impossible to
debug.  It happens both on radeon and amdgpu, on wine and wine-staging and on
GCN 1.0 and polaris.

I gave up and am using dxvk now.  The fps is lower and I have to compile
llvm-svn and mesa-git, but at least I don't have to hard-reset.  Also, since
v1.0, there is a lot less stuttering.

**Comment by gemantzu** [comment]
Yup, since they cut down dx9, the game only runs on dx11 and as it seems, this is not viable atm for poe. So dxvk it is, others on reddit told me that they had some success with it.
